File: AINews.Domain/Entities/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AINews.Domain.Entities
{
    public class Event
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime EventDate { get; set; }
        public string Location { get; set; }
        public string ImageUrl { get; set; }

        // Event Cateogry relationships 
        public Guid CategoryId { get; set; }
        public EventCategory EventCategory { get; set; }

        public string CreatedById { get; set; } 
    }
}

File: ainews.web/src/api/event.ts
import { api } from "./client";
export type EventCategory = { id: string; name: string };
export type EventSummary = { id:string; title:string; imageUrl?:string; description:string; createdByName?:string; createdById:string; eventDate:string; location?:string; categoryId:string };
export type EventDetail = EventSummary;

export const listCategories = () => api.get<EventCategory[]>("/api/EventCategory/all").then(r=>r.data);
export const listEvents = (categoryId?:string) =>
  api.get<EventSummary[]>("/api/Event/all", { params: { categoryId }}).then(r=>r.data);
export const getEvent   = (id:string) => api.get<EventDetail>(`/api/Event/${id}`).then(r=>r.data);
export const createEvent = (body: {title:string;description:string;imageUrl?:string;location?:string;eventDate:string;categoryId:string; }) =>
  api.post("/api/Event", body).then(r=>r.data);
